bin/mango-notebook

```diff
@@ -47,29 +86,55 @@ fi
 # get Mango jar
 MANGO_CLI_JAR=$(${SOURCE_DIR}/find-mango-assembly.sh)
 
-SPARK_SUBMIT=$(${SOURCE_DIR}/find-spark.sh)
-echo "Using spark-submit=$SPARK_SUBMIT" 1>&2
-
-if [ -z "$SPARK_HOME" ]; then
-  PYSPARK=$(which pyspark || echo)
+# get pyspark path
+if [[ -z "${SPARK_HOME}" ]]; then
+  wrapper_path=$(get_program_path spark-script-wrapper.sh false)
+  if [[ -f "${wrapper_path}" ]]; then
+    . ${wrapper_path}
+    PYSPARK=$(find_script pyspark)
+  else
+    PYSPARK=$(get_program_path pyspark)
+  fi
+  SPARK_HOME=${PYSPARK/bin\/pyspark}
 else
-  PYSPARK="$SPARK_HOME"/bin/pyspark
+  PYSPARK="${SPARK_HOME}"/bin/pyspark
 fi
-if [ -z "$PYSPARK" ]; then
-  echo "SPARK_HOME not set and spark-shell not on PATH; Aborting."
+
+if [[ ! -f "${PYSPARK}" ]]; then
+  print_err "pyspark was not found on PATH and SPARK_HOME; Aborting."
   exit 1
 fi
 
-export PYSPARK_PYTHON=$(which python || echo)
-export PYSPARK_DRIVER_PYTHON=jupyter
-export PYSPARK_DRIVER_PYTHON_OPTS="notebook $NOTEBOOK_ARGS"
+# Use standard PYSPARK environement variables
+if [[ -z ${PYSPARK_PYTHON} ]]; then
+  export PYSPARK_PYTHON=$(get_program_path python)
+fi
+
+if [[ -z ${PYSPARK_DRIVER_PYTHON} ]]; then
+  export PYSPARK_DRIVER_PYTHON=jupyter
+fi
+
+if [[ -z ${PYSPARK_DRIVER_PYTHON_OPTS} ]]; then
+  export PYSPARK_DRIVER_PYTHON_OPTS="notebook ${NOTEBOOK_ARGS}"
+else
+  export PYSPARK_DRIVER_PYTHON_OPTS="${PYSPARK_DRIVER_PYTHON_OPTS} ${NOTEBOOK_ARGS}"
+fi
+
+if [[ -d "${SPARK_HOME}/python/lib" ]]; then
+  PY4J_ZIP_ALL_VERSIONS=( ${SPARK_HOME}/python/lib/py4j-+([0-9]).+([0-9]).+([0-9])-src.zip )
+  # use the latest version
+  PY4J_ZIP="${PY4J_ZIP_ALL_VERSIONS[-1]}"
+  if [[ ! -e ${PY4J_ZIP} ]]; then
+    print_err "${PY4J_ZIP} is not found; Aborting."
+    exit 1
+  fi
+fi
 
-PY4J_ZIP="$(ls -1 "${SPARK_HOME}/python/lib" | grep py4j)"
 export PYTHONPATH=${SPARK_HOME}/python:${SPARK_HOME}/python/lib/${PY4J_ZIP}:${PYTHONPATH}
 
 ${PYSPARK} \
   --conf spark.serializer=org.apache.spark.serializer.KryoSerializer \
   --conf spark.kryo.registrator=org.bdgenomics.mango.serialization.MangoKryoRegistrator \
   --jars ${MANGO_CLI_JAR} \
   --driver-class-path ${MANGO_CLI_JAR} \
-  "${SPARK_ARGS[@]}"
+  ${SPARK_ARGS[@]}
```

Comment:
Why are the quotes removed?

Reply:
I thought it was better to expand parameters.
I show below 2 use case tell me which is expected, maybe I did a mistake.
before to start a look at the test function:
```bash
test_param(){  
  echo "number of provided parameter: $#"
  echo "parameter 1: $1"
  echo "value 1: $2"
  echo "parameter 2: $3"
  echo "value 2: $4"
}
```

In our case `SPARK_ARGS` is an array defined as:
```bash
 SPARK_ARGS=( "--master yarn" "--deploy-mode cluster" )
```

- **case 1** call `test_param` using quote while expanding `SPARK_ARGS` array
```bash
test_param  "${SPARK_ARGS[@]}"
number of provided parameter: 2
parameter 1: --master yarn
value 1: --deploy-mode cluster
parameter 2:
value 2:
```
- **case 2** call `test_param` without quote while expanding `SPARK_ARGS` array
```bash
test_param  ${SPARK_ARGS[@]}
number of provided parameter: 4
parameter 1: --master
value 1: yarn
parameter 2: --deploy-mode
value 2: cluster
```

What is the expected behavior?

thanks @akmorrow13